applications/nrf_desktop/src/modules/ble_bond.c

```diff
@@ -870,7 +910,9 @@ static void config_set(const uint8_t opt_id, const uint8_t *data, const size_t s
 	ARG_UNUSED(data);
 	ARG_UNUSED(size);
 
-	if (state != STATE_IDLE) {
+	if ((state != STATE_IDLE) &&
+		(!IS_ENABLED(CONFIG_DESKTOP_BLE_DONGLE_PEER_ERASE_BOND_CONF_CHANNEL)
```

Comment:
The condition is wrong now. If we are e.g. in peer erase state, we would not report busy state and we will try to perform operation which is not proper.
Please split it into 2 separate checks relying on setting flag if possible to make it easier to understand and less bug prone.